src/pages/home/HeaderView.js

```diff
@@ -92,7 +92,7 @@ const HeaderView = (props) => {
             threeDotMenuItems.push({
                 icon: Expensicons.Checkmark,
                 text: props.translate('newTaskPage.markAsDone'),
-                onSelected: () => Task.completeTask(props.report.reportID, title),
+                onSelected: () => Task.completeTask(props.report.reportID, props.report.reportName, props.report.lastMessageText),
```

Comment:
```suggestion
                onSelected: () => Task.completeTask(props.report.reportID, title, props.report.lastMessageText),
```
I don't see why this change is required, `title` is not necessary `reportName`.

Reply:
More info on why I changed that [here](https://github.com/Expensify/App/pull/20373#issuecomment-1585071638)
 
> title is not necessary reportName

I think we can assume the `title` is always the report name for Tasks specifically [because of this logic](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.js#L995-L997), no?

Reply:
> I think we can assume the title is always the report name

Yes, I think so, but this does not explain why you we are changing the title if we expect them to be the same.

I think the bug you linked is due the passing the wrong report to `getReportName` we are passing the parent report which is a chat report thus the name is the participant's email/display name. I think we may need to pass the report itself (task report).

Reply:
> I think the bug you linked is due the passing the wrong report to getReportName we are passing the parent report which is a chat report thus the name is the participant's email/display name. I think we may need to pass the report itself (task report).

I agree! Pretty sure I tried to do that and failed but **I'll give another look and try again**!

Reply:
I'll take some more time to follow up to this, after I merged `main` a bunch of errors started showing for me (when opening the app, reports and creating tasks which says `There was an error assigning this task, please try another assignee.Confirm task` forever)

I'll shift focus to another P.R. then get back to this tomorrow.

Reply:
`title` can be different based on this logic https://github.com/Expensify/App/blob/926277a188c9952ec567ab47254b1c6e45a2de38/src/pages/home/HeaderView.js#L79-L80

It looks like maybe we have some dup logic when setting `reportHeaderData` since `ReportUtils.getReportName` also does checks for task reports but i'm not 100% sure, just what i'm seeing from a quick look.